cache/interface.go

```diff
@@ -0,0 +1,63 @@
+package cache
+
+import (
+	"context"
+	"os"
+	"time"
+
+	l "github.com/RedHatInsights/sources-api-go/logger"
+	redisClient "github.com/RedHatInsights/sources-api-go/redis"
+	valkeyClient "github.com/RedHatInsights/sources-api-go/valkey"
+	"github.com/go-redis/redis/v8"
+)
+
+// Client interface defines the unified cache operations
+// This allows us to use either Redis or Valkey transparently
+type Client interface {
+	Ping(ctx context.Context) *redis.StatusCmd
+	Set(ctx context.Context, key string, value interface{}, expiration time.Duration) *redis.StatusCmd
+	Get(ctx context.Context, key string) *redis.StringCmd
+	Del(ctx context.Context, keys ...string) *redis.IntCmd
+	Exists(ctx context.Context, keys ...string) *redis.IntCmd
+	Expire(ctx context.Context, key string, expiration time.Duration) *redis.BoolCmd
+	TTL(ctx context.Context, key string) *redis.DurationCmd
+	RPush(ctx context.Context, key string, values ...interface{}) *redis.IntCmd
+	BLPop(ctx context.Context, timeout time.Duration, keys ...string) *redis.StringSliceCmd
+	Close() error
+}
+
+// Nil constant for checking nil values from cache operations
+const Nil = redis.Nil
+
+var client Client
+
+// Init initializes the cache client based on environment
+// Uses Valkey if VALKEY_CACHE_HOST is set (production), otherwise Redis (development)
```

Comment:
I'm not sure I follow this: 
Why is there business logic based on the environment? Shouldn't it be agnostic and just configurable regardless of the environment? I hope we don't go for the "I only test in production" approach for Valkey's troubleshooting/debugging events 😁